smithy-model/src/main/java/software/amazon/smithy/model/shapes/ShapeIndex.java

```diff
@@ -216,18 +186,21 @@ public Builder addShape(Shape shape) {
          * @return Returns the builder.
          */
         public Builder addShapes(ShapeIndex shapeIndex) {
-            shapeIndex.shapes().forEach(this::addShape);
+            shapeMap.putAll(shapeIndex.shapeMap);
             return this;
         }
 
         /**
          * Adds a collection of shapes to the builder.
          *
          * @param shapes Collection of Shapes to add.
+         * @param <S> Type of shape being added.
          * @return Returns the builder.
          */
-        public Builder addShapes(Collection<Shape> shapes) {
-            shapes.forEach(this::addShape);
+        public <S extends Shape> Builder addShapes(Collection<S> shapes) {
+            for (Shape shape : shapes) {
+                addShapes(shape);
```

Comment:
It's just redirected in the end, but should be `addShape` not `addShapes` for singular adds.